src/firefly/java/edu/caltech/ipac/firefly/server/events/ServerEventManager.java

```diff
@@ -32,50 +32,57 @@
     private static long deliveredEventCnt;
 
 
-
-    public static void fireAction(FluxAction action) { fireJsonAction(action.toString(), ServerEvent.Scope.SELF,null);
-    }
-
-    public static void fireAction(FluxAction action, String channel) {
-        fireJsonAction(action.toString(), ServerEvent.Scope.CHANNEL, channel);
+    /**
+     * Send this action to the calling client, i.e Scope.SELF.
+     * @param action
+     */
+    public static void fireAction(FluxAction action) {
+        fireAction(action, ServerEvent.Scope.SELF);
     }
 
+    /**
+     * Send this action to the calling client, i.e Scope.SELF.
+     * @param action
+     */
     public static void fireAction(FluxAction action, ServerEvent.Scope scope) {
-        fireJsonAction(action.toString(), scope, null);
+        fireJsonAction(action.toString(), scope);
     }
 
-    public static void fireJsonAction(String actionStr, String channel) {
-        fireJsonAction(actionStr, ServerEvent.Scope.CHANNEL, channel);
+    /**
+     * Send this action to a specific client based on the given target.
+     * @param action
+     * @param target
+     */
+    public static void fireAction(FluxAction action, ServerEvent.EventTarget target) {
+        fireJsonAction(action.toString(), target);
     }
 
-    public static void fireJsonAction(String actionStr, ServerEvent.Scope scope, String channel) {
-        ServerEvent sev;
-        if (channel!=null) {
-            ServerEvent.EventTarget t= new ServerEvent.EventTarget(ServerEvent.Scope.CHANNEL,null,channel);
-            sev = new ServerEvent(Name.ACTION, t, ServerEvent.DataType.JSON, actionStr);
-        }
-        else {
-            sev = new ServerEvent(Name.ACTION, scope, ServerEvent.DataType.JSON, actionStr);
-        }
-
+    /**
+     * Send this JSON string action to the client based on the given scope
+     * @param actionStr
+     * @param scope
+     */
+    public static void fireJsonAction(String actionStr, ServerEvent.Scope scope) {
```

Comment:
This is intended.  The idea is that you should be able to send an action to self(connId), user, channel, or world without having to create a target.